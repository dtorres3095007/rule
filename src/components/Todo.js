import React from 'react'

export default function Todo({ todo }){
    const {id, title, completed} = todo;
    const h1 = <h1><p className='algo'>{title}</p></h1>;
    const text = completed ? <strike>{h1}</strike>: h1;
    return <div data-testid={`todo-${id}`}>{text}</div>
}