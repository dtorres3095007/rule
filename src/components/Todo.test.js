import { render, screen, cleanup} from '@testing-library/react'
import Todo from './Todo'

afterEach(()=> {
     cleanup();
});

test('Should render non-completed todo', () => {
    const todo =    {id : 1, title : 'Wash dishes', completed : false};
    render(<Todo todo={todo}/>);
    const todoElemnt = screen.getByTestId('todo-1');
    expect(todoElemnt).toBeInTheDocument();
    expect(todoElemnt).toHaveTextContent('Wash dishes');
    expect(todoElemnt).not.toMatchSnapshot('<strike>');
});

test('Should render completed todo', () => {
    const todo =    {id : 2, title : 'Wash car', completed : true};
    render(<Todo todo={todo}/>);
    const todoElemnt = screen.getByTestId('todo-2');
    expect(todoElemnt).toBeInTheDocument();
    expect(todoElemnt).toHaveTextContent('Wash car');
    expect(todoElemnt).toMatchSnapshot('<strike>');
});