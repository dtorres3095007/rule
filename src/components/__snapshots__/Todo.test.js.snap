// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render completed todo: <strike> 1`] = `
<div
  data-testid="todo-2"
>
  <strike>
    <h1>
      Wash car
    </h1>
  </strike>
</div>
`;
